### DISCUSSION TITLE
Contributing back

### DISCUSSION THREAD

**Discussion by mhspelt** [vikramlearning/blazorbootstrap#825] [General]
Hi @gvreddy04 ,

We've been using Blazor.Bootstrap at our company for a while and this week, I've been making some changes to mostly the chart components. For example, we've added

* support for the secondary y-axis
* additional options for [filling](https://www.chartjs.org/docs/latest/charts/area.html#filling-modes)
* support for [chartjs-annotation](https://www.chartjs.org/chartjs-plugin-annotation/latest/)
* an UpdateValuesAsync that will only update dataset values (so the chart animates from the previous state) instead of resetting the whole config

We'd be happy to give this back. Would you be interested in receiving these changes? I now have one commit for each feature, would you like them as a single PR, or prefer a single PR rebased to the main branch for each feature separately?

**Comment by gvreddy04** [comment]
@mhspelt Thank you for using **BlazorBootstrap**. And thank you for reaching out to contribute back. Please create a PR for each feature with minimal changes and one example for each. This will help.

**Reply by mhspelt** [reply]
Thanks @gvreddy04 for the reply. We're preparing the PRs and will contribute them shortly.

Quick question: in our Visual Studio settings we are using slightly different formatting settings. Would you consider adding an .editorconfig file to the project that will allow contributions to match your code style, regardless of the contributors settings? At the moment I have to remember to set my formatting settings back to default before committing BlazorBootstrap, then setting them back to our company's standard before committing our own code.

**Reply by gvreddy04** [reply]
@mhspelt I'm using JetBrains ReSharper VS extension for code formatting. If you have the same license, I will give you the same settings file. That is the reason I didn't commit the file to the repo. But no worries, I'll run the formatting on top of your PR and test your code before merging it into the main branch. I'll take care of them.

**Reply by mhspelt** [reply]
> @mhspelt I'm using JetBrains ReSharper VS extension for code formatting. If you have the same license, I will give you the same settings file. That is the reason I didn't commit the file to the repo. But no worries, I'll run the formatting on top of your PR and test your code before merging it into the main branch. I'll take care of them.

We're not using ReSharper, so I'll leave the formatting to you then. Thanks!

**Comment by mhspelt** [comment]
@gvreddy04  I've created the PR's for the smaller changes that we've done. There is a bigger one, which adds support for arbitrary chart.js plugins and provides some built-in configuration for two of the most-used chart.js plugins: annotations and zoom. I still have to sort those out a bit but I'll wait to see if you're happy with the contributions so far or if you have any feedback.
